jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -23,25 +23,33 @@ class DeploymentDirectDeploy extends React.Component {
     super(props);
     this.state = {
       entityModel: false,
-      isBundle: this.props.ddData.id.indexOf('bundle') !== -1
+      isBundle: this.props.ddData.id.indexOf('bundle') !== -1,
+      loading: false
     };
   }
 
-  componentDidMount() {
-    this.props.getEntity(this.props.ddData.id, (error, data) => {
-      if (error) {
-        console.error('cannot fetch the entity:' + error);
-        return;
-      }
-      if (data.length > 0) {
-        data = data[0];
-        this.setState({
-          entityModel: this.props.makeEntityModel(data)
+  componentWillMount() {
+    this.setState({loading: true}, () => {
+      this.props.getEntity(this.props.ddData.id, (error, data) => {
+        this.setState({loading: false}, () => {
+          if (error) {
+            console.error('cannot fetch the entity:' + error);
+            return;
+          }
+          if (data && data.length > 0) {
```

Comment:
is this conditional necessary? There is nothing handling an 'else' case here so I'm assuming either we have an extra case to handle, or this isn't necessary and we can hard code the `data[0]`